### ISSUE TITLE
Isolate+DatabaseConnection.delayed query stream issue

### ISSUE THREAD

**Issue by Mike278** [simolus3/moor#2158]
When using an isolate with the `DatabaseConnection.delayed` constructor, no query is sent upon re-subscribing to the same instance of a query stream.

Relatively minimal repro compared to where it was ripped out of!
```dart
import 'dart:isolate';

import 'package:drift/drift.dart';
import 'package:drift/isolate.dart';
import 'package:drift/native.dart';
import 'package:test/test.dart';

void main() {
  driftRuntimeOptions.dontWarnAboutMultipleDatabases = true;

  for (final connectionType in ['isolate', 'isolate_delayed', 'non_isolate']) {
    test('re-listen to stream: $connectionType', () async {
      final _EmptyDb db;
      if (connectionType == 'isolate') {
        db = _EmptyDb.connect(await _connect());
      } else if (connectionType == 'isolate_delayed') {
        db = _EmptyDb.connect(DatabaseConnection.delayed(Future(_connect)));
      } else {
        db = _EmptyDb(NativeDatabase.memory(logStatements: true));
      }

      final stream = db
          .customSelect('select 1 as x')
          .map((x) => x.read<int>('x'))
          .watchSingle();

      Future<void> listenThenCancel() async {
        final result = await stream.first.timeout(
          const Duration(seconds: 2),
          onTimeout: () => fail('timed out (connectionType=$connectionType)'),
        );
        expect(result, equals(1));
      }

      await listenThenCancel();
      await pumpEventQueue();
      await listenThenCancel(); // times out here when using DatabaseConnection.delayed
    });
  }
}

Future<DatabaseConnection> _connect() async {
  final out = ReceivePort();
  final args = out.sendPort;
  await Isolate.spawn(_isolateEntrypoint, args);
  final isolate = (await out.first) as DriftIsolate;
  return isolate.connect(isolateDebugLog: true);
}

void _isolateEntrypoint(SendPort out) {
  final driftIsolate = DriftIsolate.inCurrent(() {
    final driver = NativeDatabase.memory(logStatements: true);
    return DatabaseConnection.fromExecutor(driver);
  });
  out.send(driftIsolate);
}

class _EmptyDb extends GeneratedDatabase {
  _EmptyDb.connect(DatabaseConnection connection) : super.connect(connection);

  _EmptyDb(QueryExecutor ex) : super(SqlTypeSystem.defaultInstance, ex);
  @override
  final allTables = const <TableInfo>[];
  @override
  final schemaVersion = 1;
}
```

All tests pass except for the one using DatabaseConnection.delayed, which fails like:
```
[OUT]: Request (id = 0): EnsureOpen(1, null)
[IN]: Request (id = 0): RunBeforeOpen(Instance of 'OpeningDetails', 0)
[OUT]: SuccessResponse (id = 0): null
[IN]: SuccessResponse (id = 0): true
[OUT]: Request (id = 1): StatementMethod.select: select 1 as x with []
Shell: Drift: Sent select 1 as x with args []
[IN]: SuccessResponse (id = 1): Instance of 'SelectResult'
package:test_api              fail
test\isolate_test.dart 30:28  main.<fn>.listenThenCancel.<fn>
===== asynchronous gap ===========================
dart:async                    Future.timeout
test\isolate_test.dart 28:43  main.<fn>.listenThenCancel
test\isolate_test.dart 27:36  main.<fn>.listenThenCancel
test\isolate_test.dart 37:13  main.<fn>
===== asynchronous gap ===========================
dart:async                    _asyncThenWrapperHelper
test\isolate_test.dart        main.<fn>

timed out (connectionType=isolate_delayed)
```

**Comment by simolus3** [comment]
Thanks for the very helpful report! I'm sure it must have been painful to find the minimal repro and the cause in `DatabaseConnection.delayed()`.

I have fixed this issue in f41cff5fc6ecec84d259a3528714a18dcba7d36a.

**Comment by Mike278** [comment]
Hah yeah especially because our tests weren't using the `.delayed` constructor but production was 😅. `Stream.multi` saves the day again! 

Impressively fast turnaround as always, thanks so much for your great work Simon!
